### PR TITLE
Migrate AtSync barrier refcount

### PR DESCRIPTION
**Pull request by rbuch**
Fixes race condition where CkMigratable calls AtSync before migrating, causing a hang on destination PE.

### CODE REVIEW

**Review by epmikida — APPROVED** [review]

**Review by stwhite91 — DISMISSED** [review]

**Review by stwhite91 — APPROVED** [review]

**Comment by rbuch** [comment]
Please merge via rebase.

**Review by epmikida — DISMISSED** [review]
Just two small name changes, otherwise lgtm

**Review by evan-charmworks — APPROVED** [review]

File: src/ck-core/ckmigratable.h [hunk 1]
```diff
@@ -103,15 +103,15 @@ class CkMigratable : public Chare {
   void ResumeFromSyncHelper();
 public:
   void ReadyMigrate(bool ready);
-  void ckFinishConstruction(void);
+  void ckFinishConstruction(int refcount = -1);
```
[the remark this answers is not in the document]
Comment (rbuch) [reply src/ck-core/ckmigratable.h]: Fixed.

File: src/ck-core/ckmigratable.h [hunk 2]
```diff
@@ -103,15 +103,15 @@ class CkMigratable : public Chare {
...
   void setMigratable(int migratable);
   void setPupSize(size_t obj_pup_size);
 #else
   void AtSync(int waitForMigration=1) { ResumeFromSync();}
   void setMigratable(int migratable)  { }
   void setPupSize(size_t obj_pup_size) { }
 public:
-  void ckFinishConstruction(void) { }
+  void ckFinishConstruction(int refcount) { }
```
[the remark this answers is not in the document]
Comment (rbuch) [reply src/ck-core/ckmigratable.h]: Fixed.

**Comment by evan-charmworks** [comment]
@rbuch Do you still want this to be merged by rebase? Will GitHub take care of the fixup commit automatically, or will that require a force-push?

**Comment by rbuch** [comment]
> @rbuch Do you still want this to be merged by rebase? Will GitHub take care of the fixup commit automatically, or will that require a force-push?

I don't think it takes care of fixup commits, I'll rebase this and force push manually.

**Review by evan-charmworks — APPROVED** [review]

**Review by epmikida — APPROVED** [review]

**Pull request merged**